### PR TITLE
fix metric names with "@" signs for lightgbm autolog

### PR DESCRIPTION
**Pull request by mengchendd**
## What changes are proposed in this pull request?

Fix same bug for lightgbm https://github.com/mlflow/mlflow/issues/5391 .

## How is this patch tested?

(Details)

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)
lightgbm autologging: Support lightgbm metrics with "@" in the name like "ndcg@2" by renaming to "ndcg_at_2"

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/lightgbm.py [hunk 1]
```diff
@@ -371,12 +371,27 @@ def __init__(self, lgb_model):
     def predict(self, dataframe):
         return self.lgb_model.predict(dataframe)
 
+def _patch_metric_names(metric_dict):
+    # XGBoost provides some metrics with "@", e.g. "ndcg@3" that are not valid MLflow metric names
```
Comment (harupy) [inline mlflow/lightgbm.py:375]: Can we fix the comment?

**Comment by harupy** [comment]
Thanks for the PR! Can we add tests?

File: tests/lightgbm/test_lightgbm_autolog.py [hunk 2]
```diff
@@ -387,6 +387,60 @@ def record_metrics_side_effect(self, metrics, step=None):
     assert "train-multi_logloss" in patched_metrics_data
 
 
+@pytest.mark.large
+def test_lgb_autolog_batch_metrics_logger_logs_ranking_metrics(bst_params, train_set):
```
Comment (harupy) [inline tests/lightgbm/test_lightgbm_autolog.py:391]: Can we simplify the test?

Comment (harupy) [reply tests/lightgbm/test_lightgbm_autolog.py:391]: I think we can do the same thing as `test_xgb_autolog_atsign_metrics`:

1. Enable autologging
2. Train
3. Fetch the latest test run logged in the previous step.
4. Extract the metrics from the run and confirm `@` was converted to `_at_`.

**Review by dbczumar — APPROVED** [review]
LGTM once @harupy 's comments have been addressed. Thank you so much for your contribution, @mengchendd !

File: tests/lightgbm/test_lightgbm_autolog.py (hunk 2 above)
Comment (harupy) [reply tests/lightgbm/test_lightgbm_autolog.py]: I pushed a commit to simplify the test.

**Pull request merged**